crates/node/p2p/src/rpc/request.rs

```diff
@@ -227,6 +227,9 @@ impl P2pRpcRequest {
             .map(|id| (id, gossip.connection_gate.connectedness(&id)))
             .collect::<HashMap<PeerId, Connectedness>>();
 
+        // Clone the ping map
+        let pings = gossip.ping.clone();
```

Comment:
Not a blocker, but this map can get _huge_ after the node has been online for a while. Let's `Arc` it and ensure that it's pruned as peers disconnect?